esmvaltool/cmorizers/data/formatters/datasets/esacci_ozone.py

```diff
@@ -0,0 +1,147 @@
+        cube.units = 'DU'
+        cube = cube * 2.24115e-5
+        cube.units = 'm'
+    return cube
+
+
+def _extract_variable(short_name, var, cfg, filename, year, month, out_dir):
+    """Extract variable, add time coordinate, and scalar longitude."""
+
+    mip = var['mip']
+    cmor_info = cfg['cmor_table'].get_variable(mip, short_name)
+
+    cubes = iris.load(filename)
+
+    cube = _convert_units(cubes, short_name, var)
+
+    logger.info("Checking CMOR info for %s: %s", short_name, cmor_info)
+    if cmor_info is None:
+        raise ValueError(f"CMOR info for {short_name} in MIP {mip} not found!")
+
+    year = year
+    month = month
+    day = 15  # Mid-month
+    time_units = Unit("days since 1950-01-01")
+    time_points = time_units.date2num(datetime(year, month, day))
+    print(f"Filename: {filename}, Extracted Year: {year}, Month: {month}")
+
+    # Add time coordinate to cube.
+    time_coord = iris.coords.DimCoord(
+        time_points,
+        var_name='time',
+        standard_name='time',
+        long_name='time',
+        units=time_units)
+    time_coord.guess_bounds(monthly=True)
+    cube.add_aux_coord(time_coord, ())
+    cube = iris.util.new_axis(cube, time_coord)
+
+    # Add longitude coordinate to cube only for o3.
+    if short_name == 'o3':
+        lon_coord = iris.coords.DimCoord(
+            [180.0],
+            bounds=[[0.0, 360.0]],
+            var_name='lon',
+            standard_name='longitude',
+            long_name='longitude',
+            units='degrees_east')
+        cube.add_aux_coord(lon_coord, ())
+        cube = iris.util.new_axis(cube, lon_coord)
+        cube.transpose([1, 3, 2, 0])
+    fix_var_metadata(cube, cmor_info)
+    cube = fix_coords(cube)
+    set_global_atts(cube, cfg['attributes'])
+    return cube
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorization process."""
+    glob_attrs = cfg['attributes']
+    if not start_date:
+        start_date = datetime(1984, 1, 1)
+    if not end_date:
+        end_date = datetime(2023, 12, 31)
+
+    for var_name, var in cfg['variables'].items():
+
+        all_data_cubes = []
+        glob_attrs['mip'] = var['mip']
+        for year in range(start_date.year, end_date.year + 1):
+            for month in range(1, 13):
+                date_str = f"{year}{month:02}"  # YYYYMM format
+
+                # Create glob pattern, handling prefixes:
+                fname_pattern = f"*{date_str}-{var['filename']}"
+                fname = glob.glob(os.path.join(in_dir, fname_pattern))
+
+                if not fname:
+                    fname = glob.glob(os.path.join(in_dir, fname_pattern))
+                    if not fname:
+                        raise ValueError(
+                            "No file found for %s in %s-%02d", var_name, year,
+                            month
+                        )
+
+                filename = fname[0]  # Take the first matching file (if any)
```

Comment:
```suggestion
                filename = Path(in_dir) / f"{date_str}-{var['filename']}"
                if not filename.is_file():
                    raise ValueError(
                        "No file found for %s in %s-%02d",
                        var_name,
                        year,
                        month,
                    )
```

I don't think you need `glob` here, the filename should be exactly determined by the date string and the file name given in the config file. You need to `from pathlib import Path` for this.